src/main/java/io/reactivex/internal/operators/flowable/FlowableToListSingle.java

```diff
@@ -47,7 +48,7 @@ public FlowableToListSingle(Publisher<T> source, Callable<U> collectionSupplier)
     protected void subscribeActual(SingleObserver<? super U> s) {
         U coll;
         try {
-            coll = collectionSupplier.call();
+            coll = ObjectHelper.requireNonNull(collectionSupplier.call(), "The collectionSupplier returned a null collection. Null values are generally not allowed in 2.x operators and sources.");
```

Comment:
Out of curiosity, is there a place where null values _are_ allowed? The use of "generally" seems to imply such.